Add total mapping limit returned error message (#149768)
elastic/elasticsearch: docs/reference/elasticsearch/index-settings/mapping-limit.md

$$$total-fields-limit$$$
`index.mapping.total_fields.limit` {applies_to}`serverless: all`

> :   The maximum number of fields in an index. Field and object mappings, field aliases, and mapped runtime fields all count towards this limit. The default value is `1000`. Beyond this limit, Elasticsearch returns the error `Limit of total fields [X] in index [Y] has been exceeded`.

Author: ```suggestion
:   The maximum number of fields in an index. Field and object mappings, field aliases, and mapped runtime fields all count towards this limit. The default value is `1000`. Beyond this limit, Elasticsearch returns the error `Limit of total fields [X] has been exceeded`.
```

Simplify error to match more common prior per [search results](https://github.com/search?q=repo%3Aelastic%2Felasticsearch%20%22Limit%20of%20total%20fields%22&type=code)

> :   The maximum number of fields in an index. Field and object mappings, as well as field aliases count towards this limit. Mapped runtime fields count towards this limit as well. The default value is `1000`. Attempting to surpass this limit will return error `Limit of total fields [X] in index [Y] has been exceeded`.

Reviewer: ```suggestion
:   The maximum number of fields in an index. Field and object mappings, field aliases, and mapped runtime fields all count towards this limit. The default value is `1000`. Beyond this limit, Elasticsearch returns the error `Limit of total fields [X] in index [Y] has been exceeded`.
```

I don't think that error message is correct and wording could be tighter. I added the specific error message.

I consolidated messy preexisting sentences about what counts towards the field limit
